modules: openthread: Fix network namespace API usage

jukkar opened the pull request:
Some of the socket calls still used POSIX symbols, replace these by native Zephyr network API calls.

Fixes #100108

jukkar committed:
modules: openthread: Fix network namespace API usage

Some of the socket calls still used POSIX symbols, replace these
by native Zephyr network API calls.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

edmont reviewed (dismissed):
Sorry, closed by accident.

jukkar committed:
tests: net: all: Make sure OpenThread tests are run

The platform_allow line in common: overrides the setting in
individual tests, so add the platform_allow line to each test
separately so that OpenThread tests are run with proper platform.

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>

jukkar committed:
tests: net: all: Increase CONFIG_MAX_THREAD_BYTES to 5

This is done to avoid this error

  Too many thread objects (38)
  Increase CONFIG_MAX_THREAD_BYTES to 5

Signed-off-by: Jukka Rissanen <jukka.rissanen@nordicsemi.no>
